### PR TITLE
[PE-140] Added Source code for MASQ Demo

### PR DESCRIPTION
**Pull request by quantropi-btank**

### CODE REVIEW

File: c/src/demo_masq_ds.c [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    int len_msg = 1024;
+    unsigned char *msg = malloc(len_msg + 1);
+    if(msg)
+    {
+        /* Generating random message */
+        rand_string(msg, len_msg);
+    }
+
+    /* Initialize random seed */ 
+    seed_init();
```
Comment (quantropi-dlou) [inline c/src/demo_masq_ds.c]: That is not correct. No need to call "seed_init();"

File: c/src/demo_masq_ds.c [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+/* Customize the random callback function below according to specific needs. 
+ * Otherwise, default random CF is Quantropi 
+ */
+int32_t rand_cf(MASQ_RAND_handle rand_handle, int32_t rand_length,  uint8_t *rand_num){
+    
+    if (rand_num == NULL) return 0;
+    
+    if (rand_handle == NULL)
```
Comment (quantropi-dlou) [inline c/src/demo_masq_ds.c]: remove rand_handle == NULL check

File: c/src/demo_masq_ds.c [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    /* Uses provided random callback function */
+    //ds_handle_alice = MASQ_DS_init(MPPKL5, rand_cf, rand_seed_cf, NULL);  
+    
+    if(ds_handle_alice == NULL) {
+        printf("\nMASQ DS Handle init failed for Alice.\n");    
+        return MASQ_ERR_MEM_Alloc;    
+    }
+    printf("\nSuccess!\n");
+
+
```
Comment (quantropi-dlou) [inline c/src/demo_masq_ds.c:161]: generate rand seed for Bob:  gen_rand_seed_bob()  -> buffer: bob_seed[1024]
generate rand seed for Alice: gen_rand_seed_alice() -> buffer:  alice_seed[1024]

File: c/src/demo_masq_ds.c (hunk 2 above)
Comment (quantropi-dlou) [reply c/src/demo_masq_ds.c]: add comment to related line of code, passing NULL, here is the handle is NULL

File: c/src/demo_masq_ds.c [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    }
+    return 1;
+}
+
+int32_t rand_seed_cf(MASQ_RAND_handle rand_handle, uint8_t *seed, int32_t seed_length)
+{
+    if (seed == NULL) return 0;
+
+    if(rand_handle == NULL){
+        seed = seed_orig;
```
Comment (quantropi-dlou) [inline c/src/demo_masq_ds.c]: unsigned int sd = 0;
for (int i=0; i< seed_length; i++)
   sd ^= seed[i];
srand(sd);

File: c/src/demo_masq_kem.c [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        {
+            rand_num[i] = rand() & 0xff;
+        }
+    }
+    return 1;
+}
+
+
+int32_t rand_seed_cf(MASQ_RAND_handle rand_handle, uint8_t *seed, int32_t seed_length)
+{
```
Comment (quantropi-dlou) [inline c/src/demo_masq_kem.c:63]: Apply the same comments for DS

File: c/README.md [hunk 6]
```diff
@@ -1,26 +1,34 @@
...
-2. Place the `lib` folder containing QEEP and SEQUR libraries for your OS under this repository's root (`/c`)
-3. Copy the `qispace_info.h` file which contains the QiSpace API URL and access token into `/src` directory
-  - Note: If the mentioned `lib` folder and `qispace_info.h` are not available to you, please contact Quantropi Support team.
-4. Run `make install_depends` to install required libraries 
+2. Get the QiSpace SDK from Quantropi. QiSpace SDK contains SEQUR, QEEP and MASQ libraries.
+3. Unzip/Untar all the contents of the libraries in `lib` folder for your OS under this repository's folder (`/c`)
+  
+    The `lib` folder will have below files:
+  
+  ![lib, the lib folder](lib.png)
```
Comment (quantropi-dlou) [inline c/README.md:18]: please update lib.png with the 1.5 release builder

**Review by quantropi-minh — APPROVED** [review]

**Review by quantropi-minh — APPROVED** [review]

**Review by quantropi-dlou — APPROVED** [review]

**Pull request merged**